examples/dao/test/failing-path.js

```diff
@@ -0,0 +1,585 @@
+      assert.throws(
+        () => ctx.executeTx([
+          { ...executeProposalTx[0] },
+          { ...executeProposalTx[1], amountMicroAlgos: 10n }
+        ]),
+        RUNTIME_ERR1009
+      );
+
+      // recipient is wrong
+      assert.throws(
+        () => ctx.executeTx([
+          { ...executeProposalTx[0] },
+          { ...executeProposalTx[1], toAccountAddr: ctx.voterA.address }
+        ]),
+        RUNTIME_ERR1009
+      );
+
+      // from_account is wrong
+      assert.throws(
+        () => ctx.executeTx([
+          { ...executeProposalTx[0] },
+          { ...executeProposalTx[1], fromAccountAddr: ctx.voterA.address }
+        ]),
+        RUNTIME_ERR1009
+      );
+    });
+  });
+
+  describe('Withdraw Vote Deposit', function () {
+    this.beforeAll(() => {
+      // set up context
+      setUpCtx();
+
+      // optIn's
+      ctx.optInToDAOApp(ctx.proposalLsig.address());
+      ctx.optInToDAOApp(ctx.voterA.address);
+      ctx.optInToDAOApp(ctx.voterB.address);
+      ctx.syncAccounts();
+
+      // add proposal
+      ctx.addProposal();
+
+      // deposit & register yes votes
+      ctx.depositVoteToken(ctx.voterA, ctx.depositLsig, minSupport + 1);
+      ctx.vote(ctx.voterA, Vote.YES, ctx.proposalLsigAcc);
+
+      // execute proposal
+      ctx.executeProposal(ctx.proposalLsigAcc);
+    });
+
+    let withdrawVoteDepositTx;
+    this.beforeEach(() => {
+      withdrawVoteDepositTx = mkWithdrawVoteDepositTx(
+        ctx.daoAppID, ctx.govTokenID, ctx.voterA.account, ctx.depositLsig, 5
+      );
+
+      // set current time after voting over
+      ctx.runtime.setRoundAndTimestamp(10, votingEnd + 10);
+    });
+
+    it('should reject withdrawal if voting is active', () => {
+      // set current time between [votingStart, votingEnd]
+      ctx.runtime.setRoundAndTimestamp(10, votingStart + ((votingEnd - votingStart) / 2));
+
+      assert.throws(
+        () => ctx.executeTx(withdrawVoteDepositTx),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should reject withdrawal if fees is not paid by tx0', () => {
+      assert.throws(
+        () => ctx.executeTx([
+          { ...withdrawVoteDepositTx[0] },
+          { ...withdrawVoteDepositTx[1], payFlags: { totalFee: 1000 } }
+        ]),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should reject withdrawal if 2nd transaction not included', () => {
+      assert.throws(
+        () => ctx.executeTx({ ...withdrawVoteDepositTx[0] }),
+        INDEX_OUT_OF_BOUND_ERR
+      );
+    });
+
+    it('should reject withdrawal trying to withdraw more than deposited', () => {
+      const origDeposit = ctx.voterA.getLocalState(ctx.daoAppID, 'deposit');
+      assert.throws(
+        () => ctx.executeTx([
+          { ...withdrawVoteDepositTx[0] },
+          { ...withdrawVoteDepositTx[1], amount: origDeposit + 5n }
+        ]),
+        'Result of current operation caused integer underflow at line 426' // -ve value handled by TEAL
+      );
```

Comment:
add 2 more test cases:
* partial withdraws should work (eg: first withdraw: 1, second withdraw: origDeposit -1)
* overflow in partial withdraw should fail:   (eg: first withdraw: 2, second withdraw: origDeposit -1)

Reply:
done